subsys/bluetooth/mesh/prov_device.c

```diff
@@ -543,17 +553,48 @@ static void prov_data(const uint8_t *data)
 	/* Ignore any further PDUs on this link */
 	bt_mesh_prov_link.expect = PROV_NO_PDU;
 
+	/* Borrow the buffer of prov link to store provision data */
+	prov_data_ptr = (void *)bt_mesh_prov_link.auth;
+	prov_data_ptr->addr = addr;
+	prov_data_ptr->flags = flags;
+	prov_data_ptr->net_idx = net_idx;
+	prov_data_ptr->iv_index = iv_index;
+	memcpy(prov_data_ptr->net_key, pdu, 16);
+	memcpy(prov_data_ptr->dev_key, dev_key, 16);
+}
+
+static void local_input_complete(void)
+{
+	if (atomic_test_bit(bt_mesh_prov_link.flags, PUB_KEY_SENT) ||
+	    atomic_test_bit(bt_mesh_prov_link.flags, OOB_PUB_KEY)) {
+		send_input_complete();
+	} else {
+		atomic_set_bit(bt_mesh_prov_link.flags, INPUT_COMPLETE);
+	}
+}
+
+static void prov_link_closed(enum prov_bearer_link_status reason)
+{
+	int err;
+	bool identity_enable;
+
+	if (!prov_data_ptr || reason != PROV_BEARER_LINK_STATUS_SUCCESS) {
+		goto reset;
+	}
+
 	/* Store info, since bt_mesh_provision() will end up clearing it */
 	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
 		identity_enable = is_pb_gatt();
 	} else {
 		identity_enable = false;
 	}
 
-	err = bt_mesh_provision(pdu, net_idx, flags, iv_index, addr, dev_key);
+	err = bt_mesh_provision(prov_data_ptr->net_key, prov_data_ptr->net_idx,
```

Comment:
Part of `prov_data_ptr` members will be located on the odd addresses due to the `packed` attribute. Information about the packed attribute will be lost within `bt_mesh_provision` since the prototype does not assume that. Access to this data might cause the potential unaligned data exception on some platforms.

Reply:
Can't the compiler recognize it?

Reply:
I always thought the compiler would do that for me, as your commit above shows.

Reply:
Indeed, it can. The compiler can recognize and generate correct code in places where it knows data attributes. If data is passed via pointer with type changing the attribute is lost. I guess I hurried up with this prototype since data are passed as values, not via a pointer. Nevertheless, the removing packed attribute is the right solution due to removing unnecessary safe data access.